app/demo-loop/diagnostic/EnergyDiagnostic.hh

```diff
@@ -118,33 +126,39 @@ void bin_energy(const celeritas::StateHostRef& states, PointersHost& pointers);
 //---------------------------------------------------------------------------//
 // EnergyDiagnostic implementation
 //---------------------------------------------------------------------------//
+/*!
+ * Construct with grid bounds and axis.
+ */
 template<MemSpace M>
-EnergyDiagnostic<M>::EnergyDiagnostic(const std::vector<real_type>& z_bounds)
-    : Diagnostic<M>()
+EnergyDiagnostic<M>::EnergyDiagnostic(const std::vector<real_type>& bounds,
+                                      Axis                          axis)
+    : Diagnostic<M>(), axis_(axis)
 {
+    CELER_EXPECT(axis != Axis::size_);
     using HostItems
         = celeritas::Collection<real_type, Ownership::value, MemSpace::host>;
 
     // Create collection on host and copy to device
-    HostItems z_bounds_host;
-    make_builder(&z_bounds_host).insert_back(z_bounds.cbegin(), z_bounds.cend());
-    z_bounds_ = z_bounds_host;
+    HostItems bounds_host;
+    make_builder(&bounds_host).insert_back(bounds.cbegin(), bounds.cend());
+    bounds_ = bounds_host;
 
     // Resize bin data
-    resize(&energy_by_z_, z_bounds_.size() - 1);
+    resize(&energy_per_bin_, bounds_.size() - 1);
```

Comment:
Should we add a `fill(real_type(0), &energy_per_bin_)` to be safe? (I noticed I didn't initialize the particle-process diagnostic counts to zero either)

Reply:
Yes indeed, I forgot this was the resize that doesn't fill.